lib/graphql/schema/scrubber.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+require "set"
+
+module GraphQL
+  class Schema
+    # Pass options to `use(SomeTracing, scrub_variables: { ... })`
+    #
+    # @api private
+    class Scrubber
+      # Create a scrubber which filters query variables.
+      #
+      # It only filters variables, not literal values in the query string.
+      #
+      # @param whitelist [Array<String, Symbol>] Argument names to _allow_
+      # @param blacklist [Array<String, Symbol>] Argument names to _hide_
+      # @param mutations [Boolean] If false, _no_ variables to mutations will be shown
+      def initialize(whitelist: nil, blacklist: nil, mutations: nil)
+        @whitelist = whitelist ? whitelist.map(&:to_s).to_set  : nil
+        @blacklist = blacklist ? blacklist.map(&:to_s).to_set : nil
+        if @whitelist && @blacklist
+          raise ArgumentError, "#{self.class.name} supports whitelist: _or_ blacklist:, but not both."
+        end
+        @mutations = mutations
+      end
+
+      SCRUBBED = "*****"
+
+      # @return [Hash]
+      def scrubbed_variables(query)
+        var_h = query.variables.to_h
+        if @mutations == false && query.mutation?
+          filter_value(var_h, scrub_all: true)
```

Comment:
that would filter out variables of other operations in the query that might not be mutations right?

Reply:
Haha, I wonder how that would behave. Let me write a couple of tests and well see 😊